fastdata/core.py

```diff
@@ -44,23 +102,121 @@ def generate(self,
+            private (bool, optional): Whether the repository is private. Defaults to False.
+            token (Optional[str], optional): The token to use to commit to the repo. Defaults to the token saved on the machine.
+            delete_files_after (bool, optional): Whether to delete files after processing. Defaults to True.
+
+        Returns:
+            tuple[str, list[dict]]: A tuple with the generated repo_id and the list of generated data dictionaries.
+        """
+        dataset_dir = Path(repo_id.replace("/", "_"))
+        dataset_dir.mkdir(parents=True, exist_ok=True)
+        data_dir = dataset_dir / "data"
+        data_dir.mkdir(exist_ok=True)
+
+        try:
+            scheduler = CommitScheduler(
+                repo_id=repo_id,
+                repo_type="dataset",
+                folder_path=dataset_dir,
+                every=commit_every,
+                private=private,
+                token=token,
+            )
+
+            readme_path = dataset_dir / "README.md"
+
+            if not readme_path.exists():
+                DatasetCard(
+                    DATASET_CARD_TEMPLATE.format(
+                        title=repo_id,
+                        model_id=self.cli.model,
+                        system_prompt=sp,
+                        prompt_template=prompt_template,
+                        sample_input=inputs[:2],
+                    )
+                ).save(readme_path)
+
+            results = []
+            total_inputs = len(inputs)
+
+            with tqdm(total=total_inputs) as pbar:
+                with concurrent.futures.ThreadPoolExecutor(
+                    max_workers=max_workers
+                ) as executor:
+                    futures = [
+                        executor.submit(
+                            self._process_input,
+                            prompt_template,
+                            schema,
+                            temp,
+                            sp,
+                            input_data,
+                        )
+                        for input_data in inputs
+                    ]
+
+                    current_file = data_dir / f"train-{uuid4()}.jsonl"
+                    for completed_future in concurrent.futures.as_completed(futures):
+                        result = completed_future.result()
+                        if result is not None:
+                            results.append(result)
+                            with scheduler.lock:
+                                self._save_results(results, current_file)
+                        pbar.update(1)
+                        if len(results) >= max_items_per_file:
+                            current_file = data_dir / f"train-{uuid4()}.jsonl"
+                            results.clear()
+        finally:
+            if delete_files_after:
+                shutil.rmtree(dataset_dir)
```

Comment:
> I also checked the local file system and nothing was created. Any ideas for what I'm doing wrong?

@AndreaFrancis @ncoop57 I think what's happening is that by default the `dataset_dir` is deleted at the end of the process and the scheduler did not have the time to upload the last result (since it's uploading once every 5 min). The workaround solution is to trigger an upload at the end (`scheduler.trigger()` and wait its completion (`.result()`) before exiting.

```suggestion
        finally:
            scheduler.trigger().result()  # force upload last result
            if delete_files_after:
                shutil.rmtree(dataset_dir)
```

Reply:
This is an example repo I got once I've added this line: https://huggingface.co/datasets/Wauplin/my_dataset/tree/main (content is faked as I did not using Claude API).